app.rb

```diff
@@ -11,6 +11,32 @@ def initialize
     @rentals = []
   end
 
+  def run
+    loop do
+      display_menu
+      option = gets.chomp
+
+      break if option == '7'
+
+      process_options(option)
+    end
+
+    puts 'Thank you for using this app!'
+  end
+
+  def process_options(option)
+    case option
+    when '1' then book_list
+    when '2' then people_list
+    when '3' then person_create
+    when '4' then new_book
+    when '5' then new_rental
+    when '6' then rental_list
+    else
+      puts 'This is not a valid option'
+    end
+  end
```

Comment:
- [ ] The role of the `main.rb` file is to start the app and to ask the user for the initial choice of options or to quit. Please move the part of the code that takes care of that to the `main.rb` file.
- [ ] The following methods in your `app.rb` file do not follow the single responsibility, YAGNI, and KISS rules. Please note that all requests for input and input validation should be placed in the `main.rb` file or in classes/methods whose responsibilities are to request or validate input.
The methods you should refactor are:
- `new_book`, `person_create`, `teacher_create`, `student_create`, `new_rental`, and `rental_list`.